src/modules/storage/LedgerStorage.ts

```diff
@@ -671,6 +729,135 @@ export class LedgerStorage extends Storages {
         });
     }
 
+    /**
+     * Puts proposal result
+     * @param block: The instance of the `Block`
+     */
+    public putProposalResult(block: Block): Promise<void> {
+        function get_proposal_result(
+            storage: LedgerStorage,
+            proposal_id: string,
+            block_validators: IProposalValidator[],
+            voting_end_height: number
+        ): Promise<Boolean> {
+            return new Promise<Boolean>(async (resolve, reject) => {
+                try {
+                    let approved: number = 0;
+                    let opposed: number = 0;
+                    let abstained: number = 0;
+                    let rejected: number = 0;
+                    let quorum: number = 0;
+                    let Qn: number = 0;
+                    let satisfyQuorum: boolean = false;
+                    let F;
+                    let result: boolean = false;
+                    let voting_result: boolean = false;
+                    let votes = await storage.getProposalVotes(proposal_id);
+                    let totoalValidators = await storage.getValidatorsAPI(null, null);
+                    let Vn = totoalValidators.length;
+                    if (votes) {
+                        for (let vote of votes) {
+                            let voter_address = vote.voter_address;
+                            let flag = true;
+                            for (let i = 0; i < block_validators.length && flag; i++) { //loop for invalidating the votes of invalid validators
+                                if (voter_address == block_validators[i].address) {
+                                    if (block_validators[i].validation_ratio == false || block_validators[i].valid_preimage == false) {
+                                        ++rejected;
+                                        flag = false;
+                                    }
+                                }
+                            }
+                            if (flag) { // if rejected skips validator's vote goes to next vote
+                                let ballot = vote.ballot_answer;
```

Comment:
```suggestion
                                let ballot = vote.ballot;
```
You should import encrypted ballots and decrypt them and save the decrypted content in ballot_answer.

Reply:
resolved